### PR TITLE
Add a stateful screen for tournaments

### PR DESCRIPTION
**Pull request by alexandrepiveteau**
This PR introduces the stateful tournament details screen. The following features are working :

- The tournament name is displayed.
- The back button works properly.
- Users can join a tournament.
- The tournament displays if the user is a participant or an admin.

### CODE REVIEW

File: mobile/src/main/java/ch/epfl/sdp/mobile/state/tournaments/StatefulTournamentDetailsScreen.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import ch.epfl.sdp.mobile.ui.tournaments.*
+
+@Composable
+fun StatefulTournamentDetailsScreen(
+    user: AuthenticatedUser,
+    reference: TournamentReference,
+    actions: TournamentDetailsActions,
+    modifier: Modifier = Modifier,
+    contentPadding: PaddingValues = PaddingValues(),
+) {
```
Comment (alexandrepiveteau) [inline mobile/src/main/java/ch/epfl/sdp/mobile/state/tournaments/StatefulTournamentDetailsScreen.kt:21]: Document `StatefulTournamentDetailsScreen`

**Review by matt989253 — APPROVED** [review]
LGTM

File: mobile/src/main/java/ch/epfl/sdp/mobile/state/tournaments/StatefulTournamentDetailsScreen.kt (hunk 1 above)
Comment (alexandrepiveteau) [reply mobile/src/main/java/ch/epfl/sdp/mobile/state/tournaments/StatefulTournamentDetailsScreen.kt:21]: (referenced in https://github.com/epfl-SDP/android/issues/340)

**Review by barmettlerl — APPROVED** [review]
Looks good for me

File: mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/Dsl.kt [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+      )
+  setContentWithLocalizedStrings(strings) {
+    PawniesTheme {
+      ProvideFacades(
+          authentication = authenticationFacade,
+          social = socialFacade,
+          chess = chessFacade,
+          speech = speechFacade,
+          tournament = tournamentFacade,
+          content = { with(environment) { content() } },
```
[the remark this answers is not in the document]
Comment (alexandrepiveteau) [reply mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/Dsl.kt:120]: It's using the `TestEnvironement` and providing it in the receiver scope, so `content()` can be called.

**Pull request merged**